packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -274,8 +325,80 @@ const DataGridProVirtualScroller = React.forwardRef<
 
   const detailPanels = getDetailPanels();
 
+  const topPinnedRows = getRows({ renderContext, rows: topPinnedRowsData });
+
+  const topPinnedRowsHeight = topPinnedRowsData.reduce((acc, value) => {
+    acc += apiRef.current.unstable_getRowHeight(value.id);
+    return acc;
+  }, 0);
+
+  const bottomPinnedRowsHeight = bottomPinnedRowsData.reduce((acc, value) => {
+    acc += apiRef.current.unstable_getRowHeight(value.id);
+    return acc;
+  }, 0);
+
+  const mainRows = getRows({
+    renderContext,
+    rowIndexOffset: topPinnedRowsData.length,
+  });
+
+  const bottomPinnedRows = getRows({
+    renderContext,
+    rows: bottomPinnedRowsData,
+    rowIndexOffset: topPinnedRowsData.length + (mainRows ? mainRows.length : 0),
+  });
+
+  const contentProps = getContentProps();
+
+  if (contentProps.style.minHeight && contentProps.style.minHeight === '100%') {
+    contentProps.style.minHeight = `calc(100% - ${topPinnedRowsHeight}px - ${bottomPinnedRowsHeight}px)`;
+  }
```

Comment:
When there're pinned columns and also a row height that doesn't adds scrollbars, the height of the pinned columns is not taking the full height available.

![image](https://user-images.githubusercontent.com/42154031/178864692-9051ffc0-60db-4c8a-b708-bc68575b7bcf.png)

Example: https://codesandbox.io/s/rowpinning-demo-mui-x-forked-9n94e1?file=/demo.tsx

Reply:
Fixed now